sqlglot/expressions.py

```diff
@@ -5295,6 +5295,32 @@ def replace_tables(expression, mapping):
         The mapped expression.
     """
 
+    def _replace_columns(node):
+        if isinstance(node, Column) and node.table:
+            # Find node's table name in mapping. It may be `db.tbl` or it may be `tbl`.
+            to_replace = None
+            for name in mapping.keys():
+                if name == node.table or name.endswith(f".{node.table}"):
+                    if to_replace is not None:
+                        raise ValueError(
+                            f"{node.table} is ambiguous in table mapping: Multiple databases with table {node.table}"
+                        )
+                    to_replace = name
+            new_name = mapping.get(to_replace)
+            if new_name:
+                new_table = to_table(
+                    new_name,
+                    **{k: v for k, v in node.args.items() if k not in ("this", "db", "catalog")},
+                )
+                return column(
+                    col=node.name,
+                    table=new_table.this if node.this else None,
+                    db=new_table.db if node.db else None,
+                    catalog=new_table.catalog if node.catalog else None,
```

Comment:
The intent of these is to preserve the structure of the original table name, but I'm pretty sure database names and catalog names can't be used in a column name, but I'm not sure.